pkg/controllers/state/statenode.go

```diff
@@ -296,6 +296,7 @@ func (in *StateNode) MarkedForDeletion() bool {
 	return in.markedForDeletion ||
 		(in.NodeClaim != nil && !in.NodeClaim.DeletionTimestamp.IsZero()) ||
 		(in.Node != nil && in.NodeClaim == nil && !in.Node.DeletionTimestamp.IsZero())
+	// || in.Taints().Has(RequireNoScheduleTaint())
```

Comment:
Is this still meant to be commented out? Or are we just handling this explicitly through mark/unmark calls in the code?

Reply:
Good question. I was thinking about adding it in, but I don't think it honestly gets any value, considering we add and remove taints whenever we mark and unmark for deletion. What are your thoughts here?